extensions/resource-deployment/src/interfaces.ts

```diff
@@ -3,3 +3,77 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 'use strict';
+
+export interface ResourceType {
+	name: string;
+	displayName: string;
+	description: string;
+	platforms: string[];
+	icon: { light: string; dark: string };
+	options: ResourceTypeOption[];
+	providers: DeploymentProvider[];
+	getProvider(selectedOptions: { option: string, value: string }[]): DeploymentProvider | undefined;
+}
+
+export interface ResourceTypeOption {
+	name: string;
+	displayName: string;
+	values: ResourceTypeOptionValue[];
+}
+
+export interface ResourceTypeOptionValue {
+	name: string;
+	displayName: string;
+}
+
+export interface DeploymentProvider {
+	notebook: string | NotebookInfo;
+	requiredTools: ToolRequirementInfo[];
+	when: string;
+}
+
+export interface NotebookInfo {
+	win32: string;
+	darwin: string;
+	linux: string;
+}
+
+export interface ToolRequirementInfo {
+	name: string;
+	version: string;
+}
+
+export enum ToolType {
+	Unknown,
```

Comment:
seems like having this list in the code maybe a limitation?

Reply:
based on the current design, we do need to implement the install/version check of a tool in the code, we are still researching and working with the PM team, if I find it can be done via json configuration I will remove the code implementation.